### PR TITLE
User story 14

### PR DESCRIPTION
**Pull request by EricLarson2020**
User will get an error message if username or password does not match, and if they are already logged in they will go to their corresponding dashboard

### CODE REVIEW

File: app/controllers/sessions_controller.rb [hunk 1]
```diff
@@ -1,20 +1,43 @@
 class SessionsController < ApplicationController
-    def new 
-    end 
+  def new
 
-    def create
```
Comment (EricLarson2020) [inline app/controllers/sessions_controller.rb:5]: Had to modify this, as I do not think we were actually finding the user before

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -1,20 +1,43 @@
 class SessionsController < ApplicationController
-    def new 
-    end 
+  def new
 
-    def create 
-        user = User.find_by(params[:id])
+    if session[:user_id]
+      user = User.find_by(id: session[:user_id])
+      flash[:notice] = "You are already logged in"
```
Comment (EricLarson2020) [inline app/controllers/sessions_controller.rb:6]: could probably use some helper method here

File: app/controllers/sessions_controller.rb [hunk 3]
```diff
@@ -1,20 +1,43 @@
 class SessionsController < ApplicationController
-    def new 
-    end 
+  def new
```
Comment (EricLarson2020) [inline app/controllers/sessions_controller.rb:2]: Need to still use the current user method

File: app/controllers/sessions_controller.rb [hunk 4]
```diff
@@ -1,20 +1,43 @@
...
-        if user.role == "admin" 
+        if user.authenticate(params[:password]) && user.role == "admin"
             flash[:notice] = "You are logged in as #{user.name}"
             redirect_to "/admin/dashboard"
-        elsif user.role == "merchant"
+        elsif user.authenticate(params[:password]) && user.role == "merchant"
             flash[:notice] = "You are logged in as #{user.name}"
             redirect_to "/merchant/dashboard"
-        else 
+        elsif user.authenticate(params[:password]) && user.role == "default"
```
Comment (whitneykidd) [inline app/controllers/sessions_controller.rb:30]: I'm glad you added in this .authenticate method 👍

File: app/controllers/sessions_controller.rb (hunk 1 above)
Comment (jhgould) [reply app/controllers/sessions_controller.rb:5]: Good catch!

**Pull request merged**